analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -130,4 +134,17 @@ private static ProgramState[] ProcessLinqEnumerable(SymbolicContext context, IIn
             },
             _ => new[] { context.State }
         };
+
+    private static ProgramState[] ProcessInformationIsNothing(SymbolicContext context, IInvocationOperationWrapper invocation) =>
+        context.State[invocation.Arguments[0].ToArgument().Value]?.Constraint<ObjectConstraint>() switch
+        {
+            ObjectConstraint constraint when constraint == ObjectConstraint.Null => new[] { context.SetOperationConstraint(BoolConstraint.True) },
+            ObjectConstraint constraint when constraint == ObjectConstraint.NotNull => new[] { context.SetOperationConstraint(BoolConstraint.False) },
+            _ when invocation.Arguments[0].TrackedSymbol() is { } argumentSymbol => new[]
```

Comment:
`argumentSymbol` must also be a reference type (Nothing is assignable to value types).

Reply:
I'd need something like this and that is not accessible via ShimLayer
```
_ when invocation.Arguments[0].ToArgument().InConversion is { Exists: true, IsReference: false, IsNullable: false  } => new[] { context.SetOperationConstraint(BoolConstraint.False) },
```
I'll make a note to do it later if we have time

Reply:
There is something like `argumentSymbol.IsReferenceType` you could use instead.

Reply:
I considered that option, but it seemed too complicated to implement given the niche of the situation. And it would not solve it for non-symbol expressions

Reply:
We should be looking at `invocation.Arguments[0].ToArgument().Value` :)

Reply:
There's line 542 
```
[DataRow("Integer?")]
```
that proves that `Integer?` behaves the same as `Object` or `Exception`